### DISCUSSION TITLE
[General] Transactions Failing but Pass on Local Fork

### DISCUSSION THREAD

**Discussion by omurovec** [zkSync-Community-Hub/zksync-developers#434] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

Trying to send a transaction through the Mainnet RPC but fails due to some unkown criteria

### What is the specific issue or error you're encountering?

Transaction fails without any details however it passes on a local mainnet fork at the same block or when sending with different tooling directly after.

The transaction fails when sending using `web3==6.15.1` python package however the *same* transaction passes when sending with `zkcast`. Other transactions also pass with the same python library.

Failing transaction: https://explorer.zksync.io/tx/0xb4c2ce6437bc50abcd6b3b37e18e38f1c1eea207879689fdb08f2b87fb7ba47a
Passing transaction: https://explorer.zksync.io/tx/0xb8ab618c7cd86635bf56435994cf62082c562516ff9b6d749698e395549d9b68

(this is a permissionless function, the sender addresses are different but this does not affect the function)

### Can you share the error messages or logs you're receiving, if any?

https://explorer.zksync.io/tx/0xb4c2ce6437bc50abcd6b3b37e18e38f1c1eea207879689fdb08f2b87fb7ba47a

### Have you made any recent changes to the contract before encountering this issue?

no

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A, the two transactions have identical data and no contract state was changed between them.

### Have you tried to isolate the problem, and if so, what were the results?

The only contributing factor which could be causing an issue would be that the timestamp is not resolving properly during execution or there is some other issue after passing the transaction request to the RPC. Note, **the same transaction at the same block number on a local fork does work**.

### What steps have you already taken to try to resolve the issue?

We've tried:
- Running the transaction on a local fork using `era_test_node`, the transaction succeeds perfectly
- Sending the transaction manually using `zkcast`, the transaction succeeds perfectly

It only fails when sending through the `web3==6.15.1` python library (occurred on multiple machines) **However, ** as noted, other transactions are executing normally using the same library. And after 12h of attempting to send the transaction it was finally submitted (https://explorer.zksync.io/tx/0xb8ab618c7cd86635bf56435994cf62082c562516ff9b6d749698e395549d9b68). Same data, same account, same contract state. It's unclear why this passed and the previous 12h of transactions failed without any error data.

### Additional Details

We have been running these bots for over 2 weeks and this is the first time we have noticed something like this happen (first noticed 12h ago)

**Comment by bxpana** [comment]
Thanks for the heads up on this. Will check with the team and get back on next steps. Good to see you @omurovec 😁

**Reply by omurovec** [reply, marked as answer]
Thanks! Just realizing now, it seems like it's due to gas being set too low.
Would be great to add a flag though in the block explorer to see this though, was quite difficult to determine the reason for failures

**Reply by dutterbutter** [reply]
Yes, keep in mind era-test-node has fixed params for L1 gas price for example. We are working on adding support for dynamic pricing to improve its accuracy. Can we consider this closed @omurovec if its resolved?
